Improving stdlib

mahyarsamani opened the pull request.

mahyarsamani committed:
cpu,stdlib: Updating StridedGen functionality.

This change updates the functionality of StridedGen to allow for
more flexible configuration of the generator. This change makes
LinearGen obsolete.

Change-Id: I0165da1f94b9bc390da1acbb53833f42b7c37f96

mahyarsamani committed:
stdlib: Interface to get monolithic memory range.

This change add the functionality to get the uninterleaved address
range from an AbstractMemory object.

Change-Id: I176c7d7b7215cc0d5ab03b90426d1253fb0bb703

powerjg commented:
Can you add some details to this PR in the description on how/why you are "improving the stdlib"?

Please give the reviewers some context so they can provide high quality reviews.

mahyarsamani committed:
stdlib: Updating dma_requestor for chi.

This change updates the configuration of dma_requestor in chi
cachehierachy. It sets two flags of `dealloc_backinv_unique` and
`dealloc_backinv_shared` to False. Since the dma_requestor does
not have an upstream cache, it does not really matter whether
they are set to true or false. However, setting them to false
ensures the unearthing of any flaws in the underlying SLICC code.

Change-Id: Ifb185eecbb6ff10657d64715e3fde020e71f3d9d

powerjg commented on src/python/gem5/components/cachehierarchies/chi/nodes/dma_requestor.py:
```diff
@@ -58,8 +58,8 @@ def __init__(self, network, cache_line_size, clk_domain: ClockDomain):
         self.dealloc_on_shared = False
-        self.dealloc_backinv_unique = True
-        self.dealloc_backinv_shared = True
+        self.dealloc_backinv_unique = False
```
Is this a bug fix?

powerjg commented on src/python/gem5/components/memory/memory.py:
```diff
@@ -170,6 +170,10 @@ def get_mem_ports(self) -> Sequence[Tuple[AddrRange, Port]]:
         return [ctrl for ctrl in self.mem_ctrl]
 
+    @overrides(AbstractMemorySystem)
+    def get_monolithic_range(self) -> int:
```
How important is this change?

powerjg reviewed (commented):
We need some updates on this change *ASAP* since the staging branch is being creating on Friday

BobbyRBruce commented:
I've cherry-picked the change I see as being of-value here to #665. I'm closing this PR as it appears to have been abandoned.
